libbeat/reader/syslog/message.go

```diff
@@ -102,13 +106,15 @@ func (m *message) setTimestampBSD(v string, loc *time.Location) {
 	}
 	if t, err := time.ParseInLocation(time.Stamp, v, loc); err == nil {
 		t = t.AddDate(time.Now().In(loc).Year(), 0, 0)
+		m.timestampSet = true
```

Comment:
You could just use `m.timestamt.IsZero()` later. It's cheap and I don't think people were using syslog in 1CE.

Reply:
Apparently I wasn't thinking straight yesterday and forgot IsZero existed 😆 . The whole point of this was to prevent us from accidentally setting the time to 1CE on the event, since that's what m.timestamp initializes to and we are now passing fields and the timestamp along, even when there's an error.